api4/status.go

```diff
@@ -123,6 +127,85 @@ func updateUserStatus(c *Context, w http.ResponseWriter, r *http.Request) {
 	getUserStatus(c, w, r)
 }
 
+func getUserSchedule(c *Context, w http.ResponseWriter, r *http.Request) {
+	c.RequireUserId()
+	if c.Err != nil {
+		return
+	}
+
+	// No permission check required
+
+	schedule, err := c.App.GetUserScheduleById(c.Params.UserId)
+	if err != nil {
+		c.Err = err
+		return
+	}
+
+	if schedule == nil {
+		c.Err = model.NewAppError("UserStatus", "api.statusSchedule.user_not_found.app_error", nil, "", http.StatusNotFound)
+		return
+	}
+
+	w.Write([]byte(schedule.ToJson()))
+}
+
+func getUserStatusSchedule(c *Context, w http.ResponseWriter, r *http.Request) {
+	c.RequireUserId()
+	if c.Err != nil {
+		return
+	}
+
+	statusSchedule := model.StatusScheduleFromJson(r.Body)
+	if statusSchedule == nil {
+		c.SetInvalidParam("statusSchedule")
+		return
+	}
+
+	// The user being updated in the payload must be the same one as indicated in the URL.
+	if statusSchedule.UserId != c.Params.UserId {
```

Comment:
I think we can safely remove this check and just rely on `SessionHasPermissionToUser` which will have the benefit of making this API endpoint useful for local mode or for users with permission to `edit_other_users`.